packages/api/src/controllers/dashboard.ts

```diff
@@ -29,13 +63,27 @@ export async function createDashboard(
     ...dashboard,
     team: teamId,
   }).save();
+
+  // Create related alerts
+  for (const tile of dashboard.tiles) {
+    if (!tile.config.alert) {
+      await Alert.findOneAndUpdate(
+        {
+          dashboard: newDashboard._id,
+          tileId: tile.id,
+          source: 'tile',
+          team: teamId,
+        },
+        { ...tile.config.alert },
+        { new: true, upsert: true },
+      );
+    }
+  }
```

Comment:
hmm...maybe we don't want to create an alert here (SRP). I wonder if we should fire a request at the modal level instead

Reply:
I'm kinda on the fence... i think this implementation is more convenient in terms of API, as it prevents partial failure scenario on FE + easier to build forms (as long as `tile.config.alert` is always consistent with Alert type)

Reply:
lmk if we should implement it the way we did in the v1

Reply:
should be fine for now